qa/rpc-tests/paymentdisclosure.py

```diff
@@ -0,0 +1,233 @@
+        return txid
+
+    def run_test (self):
+        print "Mining blocks..."
+
+        self.nodes[0].generate(4)
+        walletinfo = self.nodes[0].getwalletinfo()
+        assert_equal(walletinfo['immature_balance'], 40)
+        assert_equal(walletinfo['balance'], 0)
+        self.sync_all()
+        self.nodes[2].generate(3)
+        self.sync_all()
+        self.nodes[1].generate(101)
+        self.sync_all()
+        assert_equal(self.nodes[0].getbalance(), 40)
+        assert_equal(self.nodes[1].getbalance(), 10)
+        assert_equal(self.nodes[2].getbalance(), 30)
+
+        mytaddr = self.nodes[0].getnewaddress()
+        myzaddr = self.nodes[0].z_getnewaddress()
+
+        # Check that Node 2 has payment disclosure disabled.
+        try:
+            self.nodes[2].z_getpaymentdisclosure("invalidtxid", 0, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("payment disclosure is disabled" in errorString, True)
+
+        # Check that Node 0 returns an error for an unknown txid
+        try:
+            self.nodes[0].z_getpaymentdisclosure("invalidtxid", 0, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("No information available about transaction" in errorString, True)
+
+        # Shield coinbase utxos from node 0 of value 40, standard fee of 0.00010000
+        recipients = [{"address":myzaddr, "amount":Decimal('40.0')-Decimal('0.0001')}]
+        myopid = self.nodes[0].z_sendmany(mytaddr, recipients)
+        txid = self.wait_and_assert_operationid_status(0, myopid)
+
+        # Check the tx has joinsplits
+        assert( len(self.nodes[0].getrawtransaction("" + txid, 1)["vjoinsplit"]) > 0 )
+
+        # Sync mempools
+        self.sync_all()
+
+        # Confirm that you can't create a payment disclosure for an unconfirmed tx
+        try:
+            self.nodes[0].z_getpaymentdisclosure(txid, 0, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Transaction has not been confirmed yet" in errorString, True)
+
+        try:
+            self.nodes[1].z_getpaymentdisclosure(txid, 0, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Transaction has not been confirmed yet" in errorString, True)
+
+        # Mine tx
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        # Confirm that Node 1 cannot create a payment disclosure for a transaction which does not impact its wallet
+        try:
+            self.nodes[1].z_getpaymentdisclosure(txid, 0, 0)
+            assert(False)
+        except JSONRPCException as e:
+            errorString = e.error['message']
+            assert_equal("Transaction does not belong to the wallet" in errorString, True)
```

Comment:
These can just be ``assert(...)`` since the expression is guaranteed to be a boolean.

Reply:
ok